### ISSUE TITLE
[DataLoader] Dataloader import problem for DGL 2.4

### ISSUE THREAD

**Issue by jalencato** [dmlc/dgl#7774]
## 🐛 Bug

In the release note we see for DGL 2.4.0, we said ``Users are recommended to call dgl.distributed.DistNode/EdgeDataLoader though dgl.dataloading.DistNode/EdgeDataLoader is still available``. However when we were trying to load DistDataLoader from dgl.dataloading, we met an error. May we add it back to support the backward compatabilities? 

## To Reproduce

``
from dgl.dataloading import DistDataLoader
``

## Environment

 - DGL Version (e.g., 1.0): 2.4.0
 - Backend Library & Version (e.g., PyTorch 0.4.1, MXNet/Gluon 1.3): Torch 2.3.0
 - OS (e.g., Linux): Amazon Linux
 - How you installed DGL (`conda`, `pip`, source): pip
 - Python version: 3.11
 - CUDA/cuDNN version (if applicable): 12.1
 - GPU models and configuration (e.g. V100):
 - Any other relevant information:

**Comment by mfbalin** [comment]
@Rhett-Ying

**Comment by Rhett-Ying** [comment]
@jalencato `DistDataLoader` is part of `distributed` and mainly for internal use. you should always import it via `from dgl.distributed import DistDataLoader`. For backward compatibility, we just make `from dgl.dataloading import DistNode/EdgeDataLoader` works.

**Comment by Rhett-Ying** [comment]
As `DistDataLoader` was not imported into `dgl.dataloading` on purpose, I recommend to update user code to align with latest DGL.

**Comment by classicsong** [comment]
So dgl.dataloading.DistDataLoader is deprecated in DGl 2.4?

**Comment by mfbalin** [comment]
> So dgl.dataloading.DistDataLoader is deprecated in DGL 2.4?

It is simply moved from `dgl.dataloading` to `dgl.distributed` is my understanding.

**Comment by Rhett-Ying** [comment]
> So dgl.dataloading.DistDataLoader is deprecated in DGL 2.4?

`DistDataLoader` **was and is** defined within `dgl.distributed`. And `DistNode/EdgeDataLoader` which inherited from `DistDataLoader` was defined within `dgl.dataloading` which is now moved to `dgl.distributed` while keeps the backward compatibility. `dgl.dataloading.DistDataLoader` was indirectly imported when defining `DistNode/EdgeDataLoader`. I'd like to call it as a **side effect feature**. As I've moved the real definition of `DistNode/EdgeDataLoader` to `dgl.distributed`, `DistDataLoader` is not available within `dgl.dataloading`.

In short, even before DGL 2.4, anyone that wants to use `DistDataLoader` is supposed to call `dgl.distributed.DistDataLoader` instead of `dgl.dataloading.DistDataLoader`.

**Comment by Rhett-Ying** [comment]
let me add a fix for the backward incompatible issue

**Comment by frozenbugs** [comment]
We should keep backward compatibility when rename, if `from dgl.dataloading import DistDataLoader` works pre-2.4. It should work in 2.4 as well with a deprecation warning. We will release a 2.4.1 soon and have it fixed, @jalencato can you also clean up your code in next release, we will remove the hack in the future likely EOY.

**Comment by jalencato** [comment]
Will do. Thanks.

**Comment by Rhett-Ying** [comment]
@jalencato please try with the nightly build tomorrow. If it works well, we'll patch it to DGL 2.4.1

**Comment by jalencato** [comment]
Sure, how to install the latest nightly build?

**Comment by Rhett-Ying** [comment]
refer to https://www.dgl.ai/pages/start.html
